### PR TITLE
[WIP] 616

### LINKED ISSUE #616: environment vars & cli options to specify dotfiles
In a fashion similar to `tsc --project` and `tslint --project` which allow to specify which (and where) your project configuration file to be used, it would be nice (and useful) to be able to define on the command line different clasp configuration.

## Expected Behavior

`clasp push --project dev.json`
`clasp push --project prod.json`

## Actual Behavior

(some kludgy workaround with additional folders and intricate path references)

## Steps to Reproduce the Problem

1.
1.

## Specifications

- Node version (`node -v`): 10
- Version (`clasp -v`): latest
- OS (Mac/Linux/Windows): Mac (latest)

### PR DESCRIPTION
**Pull request by PopGoesTheWza**
Fixes #616 

- [?] `npm run test` succeeds.
- [x] `npm run lint` succeeds.
- [ ] Appropriate changes to README are included in PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,7 +74,6 @@
     "multimatch": "^4.0.0",
     "normalize-newline": "3.0.0",
     "open": "^6.4.0",
-    "path": "^0.12.7",
```
Comment (PopGoesTheWza) [inline package.json]: `path` & `@types/events` packages were erroneously added and are not needed

File: src/commands/login.ts [hunk 2]
```diff
@@ -7,7 +7,6 @@ import { authorize, getLoggedInEmail } from '../auth';
 import { FS_OPTIONS } from '../files';
 import { readManifest } from '../manifest';
 import { ERROR, LOG, checkIfOnline, hasOauthClientSettings, safeIsOnline } from '../utils';
-import { google } from 'googleapis';
```
Comment (PopGoesTheWza) [inline src/commands/login.ts]: unused import

File: src/commands/login.ts [hunk 3]
```diff
@@ -16,10 +15,10 @@ import { google } from 'googleapis';
  * @param {string?} options.creds The location of credentials file.
  * @param {boolean?} options.status If true, prints who is logged in instead of doing login.
  */
-export default async (options: { localhost?: boolean; creds?: string, status?: boolean }) => {
+export default async (options: { localhost?: boolean; creds?: string; status?: boolean }) => {
```
Comment (PopGoesTheWza) [inline src/commands/login.ts]: Changes below are `tslint`fixes

File: src/commands/clone.ts [hunk 4]
```diff
@@ -1,11 +1,10 @@
-import { async } from 'rxjs/internal/scheduler/async';
```
Comment (PopGoesTheWza) [inline src/commands/clone.ts]: erroneous & unused import

File: src/commands/run.ts [hunk 5]
```diff
@@ -50,6 +50,7 @@ export default async (functionName: string, cmd: { nondev: boolean; params: stri
  */
 async function runFunction(functionName: string, params: string[], scriptId: string, devMode: boolean) {
   try {
+    // TODO: @grant what are the issues/risks of not using local auth?
```
Comment (PopGoesTheWza) [inline src/commands/run.ts]: I am not familiar with the run api. Does it requires a local auth and fails if global auth?

File: src/files.ts [hunk 6]
```diff
@@ -1,25 +1,25 @@
 import path from 'path';
-import findUp from 'find-up';
 import fs from 'fs-extra';
 import mkdirp from 'mkdirp';
 import multimatch from 'multimatch';
 import recursive from 'recursive-readdir';
-import { async } from 'rxjs/internal/scheduler/async';
```
Comment (PopGoesTheWza) [inline src/files.ts]: erroneous & unused import

**Comment by PopGoesTheWza** [comment]
@grant Not sure why is crashed on Node 8 tests.
Still, this PR is more so that you can judge and comment the design principle.

new files:
- `pathproxy.ts` the base class to handle reference to config file (file doesn't have to exist, if path points to a folder then default filename is appended, etc.)
- `conf.ts` a singleton to manage the `pathproxy` configuration files (replacing the `DOT`object)

The other changes are mostly switching for `DOT` use to `Conf`. Handling of `auth` as being either local or global can be tricky but should be fine.

Pending your feedback (take your time) I'll make the requested changes and add specific tests.

**Comment by PopGoesTheWza** [comment]
@grant approved?

**Comment by grant** [comment]
Oops. Didn't mean to approve. Just the commander change looked okay.

Looking for a smaller PR.

**Comment by PopGoesTheWza** [comment]
This PR is not meant to be merged as is. But for you to "preview" what I have in mind and give directions.
I know it will be a rather large PR (unless you choose small kludges & patches) :P

**Comment by PopGoesTheWza** [comment]
Do you want me to open a cleanup PR with the package updates (commander) and the drop of pre node 8 poly fill?

**Comment by grant** [comment]
Please create smaller PRs that only do 1 thing.
Like bump commander.

Let's continue conversation in an issue or new small PR.

**Pull request closed without merging**